progs/tests/expression/index_immediately.sy

```diff
@@ -0,0 +1,14 @@
+A :: blob {
+    a: int
+}
+
+start :: fn {
+    [1][0] + 1 <=> 2
+    { 1: 1 }[1] <=> 1
+    A { a: 1 }.a <=> 1
+    (fn -> int { 2 })() <=> 2
+    a :: [1]
+    [a][0][0] <=> 1
+    // fn { 1 }'
+    // fn { 1 }()
```

Comment:
Raise this into an issue. No one ever looks in these tests uinless they fail - and they will probably not know what they are for.